### PR TITLE
UX: refactor .d-modal to use BEM + cleanup CSS

### PR DESCRIPTION
**Pull request by chapoi**
This PR refactors the following:
* leaving all the CSS applied to the old `modal-body` classes in their respective files
* made  new clean styling for `.d-modal` and refactored the template to use the new BEM classes
  * `inner-`, `middle-`, `outer-` container classes are gone and replaced with simplified `wrapper` and `container` classes  
  * use standardised max-sizes with modifiers `-large` and `-max`
  * lighter backdrop,
  * min-width to prevent puny modals
  * other styling changes regarding padding, close button,…
* pulled out all modal overrides into a general `modal-overrides` file + cleanup of outdated CSS
* pulled out login and create account modal styling into their own file, cause it's such a big override 
* removed old general login.scss file for mobile & desktop
* only kept some remainders I don't want to touch in `app/assets/stylesheets/common/base/login.scss`

## General modals
* BEM
* modifiers for size `-large` & `-max`
* bigger min-width to avoid really puny ones
Before – After
image
image

This also means some modals will have grown a bit to the max-width of 600px – this might feel a bit jarring at first, but I think it benefits from having more standard size.

Example with bookmarks
image

* full width om mobile
Before – After
image

### Desktop
No visible difference
image

### Mobile
Full width
image

## Create account
Barely any difference, except new standard width and some more padding.
image

### Mobile
image

### CODE REVIEW

**Comment by chapoi** [comment]
@jordanvidrine @pmusaraj I'd love it if both of you could take a looksie 
it's still draft because cvx is checking some things too, but if all goes well i'd love to finally merge on monday and put this episode behind me 😜

**Comment by pmusaraj** [comment]
I can take a look only on Monday @chapoi. Also, I see some conflicts with core, they'd need to be resolved first. 

We're doing a version bump early next week, since this is a big PR, I would certainly want to avoid merging it close to a bump. (Usually, when we bump, lots of self-hosters get notified and they update right away.)

File: app/assets/stylesheets/desktop/invite-signup.scss [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.invites-show {
```
Comment (pmusaraj) [inline app/assets/stylesheets/desktop/invite-signup.scss:1]: The invites styling also feels like something that could easily be split up into a separate PR.

File: config/locales/client.en.yml [hunk 2]
```diff
@@ -1855,7 +1855,7 @@ en:
         same_as_username: "Your password is the same as your username."
         same_as_email: "Your password is the same as your email."
         ok: "Your password looks good."
-        instructions: "at least %{count} characters"
+        instructions: "At least %{count} characters."
```
Comment (pmusaraj) [inline config/locales/client.en.yml:1858]: Sorry for posting the screenshot in the wrong place, noticed a a small inconsistency in styling here: 

image

The first two descriptions' text color is different from the third.

File: plugins/chat/assets/stylesheets/common/chat-modal-thread-settings.scss [hunk 3]
```diff
@@ -0,0 +1,25 @@
+.chat-modal-thread-settings {
```
Comment (pmusaraj) [inline plugins/chat/assets/stylesheets/common/chat-modal-thread-settings.scss:1]: The styles here are also feel independent from the rest.

**Review by pmusaraj — COMMENTED** [review]
Left some comments @chapoi, overall this looks very good! 

My main concern is with risk with existing themes/plugins, I think it would be safer to split this up into smaller pieces. And also to make the class renames in a staggered fashion, to reduce breakage for themes and plugins. 

Also, note for other reviewers, locally I had to remove an import, `@import "login";` otherwise the server would fail with a SCSS error (since that file no longer exists). Will probably be fixed once the merge conflicts are resolved.

File: app/assets/javascripts/discourse/app/components/d-modal.hbs [hunk 4]
```diff
@@ -7,11 +7,7 @@
   @append={{true}}
 >
   <this.dynamicElement
-    class={{concat-class
-      "modal"
-      "d-modal"
-      (if @inline "inline-modal" "fixed-modal")
```
[the remark this answers is not in the document]
Comment (chapoi) [reply app/assets/javascripts/discourse/app/components/d-modal.hbs:13]: As far as I'm aware, the only reason it's there is for the styleguide positioning… I strongly dislike it too but just kept what we had.

File: app/assets/javascripts/discourse/app/components/d-modal.hbs [hunk 5]
```diff
@@ -21,9 +17,9 @@
     {{will-destroy this.cleanupListeners}}
     {{trap-tab preventScroll=false}}
   >
-    <div class="modal-outer-container">
```
[the remark this answers is not in the document]
Comment (chapoi) [reply app/assets/javascripts/discourse/app/components/d-modal.hbs:24]: So I put middle-container back because what you said last time that it was often used. So it's only outer- and -inner container that's really gone, but they were used less I thought.

File: app/assets/javascripts/discourse/app/components/d-modal.hbs [hunk 6]
```diff
@@ -34,33 +30,34 @@
...
+                  {{#if @subtitle}}
+                    <p class="d-modal__subtitle-text">{{@subtitle}}</p>
+                  {{/if}}
+
+                  {{yield to="belowModalTitle"}}
+                </div>
+              {{/if}}
+              {{yield to="headerBelowTitle"}}
+
               {{#if this.dismissable}}
```
[the remark this answers is not in the document]
Comment (chapoi) [reply app/assets/javascripts/discourse/app/components/d-modal.hbs:53]: okay, thanks for catching that, I'll see if I can find and fix it

File: app/assets/javascripts/discourse/app/components/d-modal.hbs [hunk 7]
```diff
@@ -80,7 +77,7 @@
             </div>
           {{/if}}
 
-          <div class={{concat-class "modal-body" @bodyClass}} tabindex="-1">
+          <div class={{concat-class "d-modal__body" @bodyClass}} tabindex="-1">
```
[the remark this answers is not in the document]
Comment (chapoi) [reply app/assets/javascripts/discourse/app/components/d-modal.hbs:80]: Hmm, maybe… I mean, both classes could be having conflicting css, in which case it won't work but will check.

File: app/assets/javascripts/discourse/app/components/modal/create-account.hbs [hunk 8]
```diff
@@ -31,244 +31,244 @@
...
-                @type="email"
-                @value={{this.model.accountEmail}}
-                disabled={{this.emailDisabled}}
-                autofocus="autofocus"
-                aria-describedby="account-email-validation"
-                aria-invalid={{this.emailValidation.failed}}
-                name="email"
-                id="new-account-email"
-                class={{value-entered this.model.accountEmail}}
+        <form id="login-form">
```
[the remark this answers is not in the document]
Comment (chapoi) [reply app/assets/javascripts/discourse/app/components/modal/create-account.hbs:34]: Ah no, probably went wrong in the refactor – shouldnt be there, thanks!

File: app/assets/stylesheets/common/base/modal.scss [hunk 9]
```diff
@@ -1,5 +1,111 @@
...
+    input {
+      width: auto;
+
+      &[type="text"] {
+        width: 100%;
+      }
+    }
+    textarea {
+      width: 100%;
+      height: 80px;
```
[the remark this answers is not in the document]
Comment (chapoi) [reply app/assets/stylesheets/common/base/modal.scss:74]: not new I think

File: app/assets/stylesheets/common/base/modal.scss [hunk 10]
```diff
@@ -532,140 +638,6 @@
   font-size: var(--font-down-1);
 }
 
-.jump-to-post-modal {
```
[the remark this answers is not in the document]
Comment (chapoi) [reply app/assets/stylesheets/common/base/modal.scss:535]: I'll be honest: I don't see how I could split that up? If I don't put in the general PR, they'll be broken…?

File: app/assets/stylesheets/desktop/invite-signup.scss (hunk 1 above)
Comment (chapoi) [reply app/assets/stylesheets/desktop/invite-signup.scss:1]: but same as I said above: it will be broken until the next PR happens then

File: app/assets/javascripts/discourse/app/components/d-modal.hbs (hunk 5 above)
[the remark this answers is not in the document]
Comment (chapoi) [reply app/assets/javascripts/discourse/app/components/d-modal.hbs:24]: interstitials plugin uses the new d-modal already, so that shouldn't cause any big issues thankfully.

File: app/assets/javascripts/discourse/app/components/d-modal.hbs (hunk 7 above)
[the remark this answers is not in the document]
Comment (chapoi) [reply app/assets/javascripts/discourse/app/components/d-modal.hbs:80]: Yeah that conflicts. I can start using !importants on the `d-modal__body` class but that's all I can think of.  Plus, the overrules in themes on `modal-body` might break things again when they conflict.

File: app/assets/javascripts/discourse/app/components/modal/create-account.hbs (hunk 8 above)
[the remark this answers is not in the document]
Comment (chapoi) [reply app/assets/javascripts/discourse/app/components/modal/create-account.hbs:34]: So I haven't changed anything to the template logic, and it seems to be like it's missing something?

```
    {{#if (and this.showLoginButtons this.site.desktopView)}}
      {{#unless this.canLoginLocal}}
        <div class="login-left-side">
          <Modal::Login::WelcomeHeader @wavingHandURL={{this.wavingHandURL}} />
        </div>
      {{/unless}}
      <div class="login-right-side">
        <LoginButtons
          @externalLogin={{this.externalLoginAction}}
          @passkeyLogin={{this.passkeyLogin}}
        />
      </div>
    {{/if}}
```

Should this have another {{if}} to render only when it has alternate login methods allowed? 

instead of hiding with css:

image

Feels more clean to me.

[the remark this answers is not in the document]
Comment (chapoi) [reply app/assets/javascripts/discourse/app/components/modal/create-account.hbs:34]: I added in this
```
{{#if this.hasAtLeastOneLoginButton}}
        <div class="login-right-side">
          <LoginButtons
            @externalLogin={{this.externalLoginAction}}
            @passkeyLogin={{this.passkeyLogin}}
          />
        </div>
      {{/if}}
```

prop seems to be what i need

File: app/assets/stylesheets/common/base/modal.scss [hunk 11]
```diff
@@ -710,26 +682,6 @@
   }
 }
 
-.topic-bulk-actions-modal {
```
[the remark this answers is not in the document]
Comment (chapoi) [reply app/assets/stylesheets/common/base/modal.scss:713]: This isn't what it's supposed to be. It's being caused by placing that modal-inner-container class back… I'll see if I can work around it.

File: config/locales/client.en.yml (hunk 2 above)
Comment (chapoi) [reply config/locales/client.en.yml:1858]: Weird but I can't repro 
image

File: app/assets/javascripts/discourse/app/components/d-modal.hbs (hunk 6 above)
[the remark this answers is not in the document]
Comment (chapoi) [reply app/assets/javascripts/discourse/app/components/d-modal.hbs]: Oh based on the meta topic, they are placed correctly 
https://meta.discourse.org/t/group-poll-results-by-user-custom-field/284710?u=chapoi

File: app/assets/stylesheets/desktop/invite-signup.scss (hunk 1 above)
Comment (pmusaraj) [reply app/assets/stylesheets/desktop/invite-signup.scss:1]: I think we can merge an invite styling PR right away. From what I can tell, it's sharing some CSS with the login modal, but it doesn't share markup, so we should split it out the same way you've done here, but as a separate PR, we can review and merge more quickly. 

It's clear to me from the screenshots (and a quick local test) that it is fixing a padding problem, and conceptually some separtion between it and login/create account modals makes 100% sense to me.

File: config/locales/client.en.yml (hunk 2 above)
Comment (pmusaraj) [reply config/locales/client.en.yml:1858]: Hmm, can't repro today either.

File: app/assets/stylesheets/desktop/invite-signup.scss (hunk 1 above)
Comment (chapoi) [reply app/assets/stylesheets/desktop/invite-signup.scss]: Moved it from this PR to https://github.com/discourse/discourse/pull/24293

File: app/assets/javascripts/discourse/app/components/d-modal.hbs (hunk 7 above)
[the remark this answers is not in the document]
Comment (chapoi) [reply app/assets/javascripts/discourse/app/components/d-modal.hbs]: discourse-user-notes still uses the old template so that won't be impacted at all.

File: app/assets/javascripts/discourse/app/components/d-modal.hbs [hunk 12]
```diff
@@ -21,87 +17,84 @@
     {{will-destroy this.cleanupListeners}}
     {{trap-tab preventScroll=false}}
   >
-    <div class="modal-outer-container">
-      <div class="modal-middle-container">
-        <div class="modal-inner-container">
-          {{yield to="aboveHeader"}}
-          {{#if
-            (and
-              (not @hideHeader)
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/d-modal.hbs:30]: Note to self: looks like we lost this new conditional when resolving merge conflicts 👀

File: app/assets/javascripts/discourse/app/components/modal/create-account.hbs [hunk 13]
```diff
@@ -31,244 +31,243 @@
...
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/modal/create-account.hbs:34]: @chapoi is there a particular reason for switching from `class="login-form"` to `id="login-form"`? If not, perhaps we should stick with `class=` to reduce the potential impact on themes/plugins?

File: app/assets/javascripts/discourse/app/components/modal/forgot-password.hbs [hunk 14]
```diff
@@ -22,8 +22,9 @@
         autocapitalize="off"
       />
     {{else}}
+      <p> {{i18n "forgot_password.invite"}}</p>
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/modal/forgot-password.hbs:25]: Is the preceding space deliberate here @chapoi?

File: app/assets/javascripts/discourse/app/components/d-modal.hbs (hunk 12 above)
Comment (pmusaraj) [reply app/assets/javascripts/discourse/app/components/d-modal.hbs:30]: Good catch, thanks!

File: app/assets/javascripts/discourse/app/components/modal/create-account.hbs (hunk 13 above)
Comment (chapoi) [reply app/assets/javascripts/discourse/app/components/modal/create-account.hbs:34]: I didn't switch, they were both in use and out of safety I just kept both

File: app/assets/javascripts/discourse/app/components/modal/forgot-password.hbs (hunk 14 above)
Comment (chapoi) [reply app/assets/javascripts/discourse/app/components/modal/forgot-password.hbs:25]: No :)

Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/app/components/modal/forgot-password.hbs:25]:
```suggestion
      <p>{{i18n "forgot_password.invite"}}</p>
```

**Review by davidtaylorhq — APPROVED** [review]
